sdk/core/azure-core/azure/core/pipeline/transport/__init__.py

```diff
@@ -48,59 +48,58 @@
 # pylint: disable=unused-import, redefined-outer-name, no-member, too-many-statements, too-many-branches
 
 
-def __dir__():
+def __dir__() -> List[str]:
     return __all__
 
 
-def __getattr__(name):
-    transport = None
+def __getattr__(name: str) -> Type[Any]:  # There is no point to be specific on the type here
     if name == "AsyncioRequestsTransport":
         try:
             from ._requests_asyncio import AsyncioRequestsTransport
 
-            transport = AsyncioRequestsTransport
+            return AsyncioRequestsTransport
```

Comment:
it was initially like that to get around the pylint error. pylint disable would also be okay here